### DISCUSSION TITLE
Archive replaces header & footer with custom Sage theme

### DISCUSSION THREAD

**Discussion by gitgudcameron** [elementor/elementor#25871] [Developers Community]
### Prerequisites

- [x] I've read the Elementor Location API docs
- [x] I have searched for similar issues in both open and closed tickets.

### Description

I have a custom Sage theme and setup Elementor Theme Location using functions (following [these instructions](https://developers.elementor.com/theme-locations-api/migrating-themes)). Header, Footer and Single custom templates using the Elementor Theme Builder displays as intended.

The issue is that when we create an archive template in Elementor Theme Builder, it seems to ignore the whole index.blade.php file which replaces the header and footer (see URL provided).

### The Index.blade.php file:

```
@extends('layouts.app')


@section('content')
@if ( ! function_exists( 'elementor_theme_do_location' ) || ! elementor_theme_do_location( 'archive' ) )

@if (!have_posts())
<div class="alert alert-warning">
  {{ __('Sorry, no results were found.', 'sage') }}
</div>
{!! get_search_form(false) !!}
@endif

@while (have_posts()) @php the_post() @endphp
@include('partials.content-'.get_post_type())

@endwhile

{!! get_the_posts_navigation() !!}

@endif
@endsection
```

See the [full code and file structure](https://github.com/AdvantageAustralia/Sage-x-Elementor-theme-support).

### How the Archive page displays with incorrect header and footer
![Screen Shot 2020-03-01 at 9 46 01 am](https://user-images.githubusercontent.com/52658950/75616435-ca004180-5ba4-11ea-8a1b-34c6f8b9b9bd.JPG)

If you can help us with this, i'd really appreciate it! Thanks in advance!

**Comment by MichaelRise** [comment]
Hey,

I have this same problem and I have a solution, but I don't know it is 100% good.
You must register 'archive' location. Not work with Elementor Fullwidth Page Layout.

functions.php
```
function myprefix_register_elementor_locations( $elementor_theme_manager ) {
	//$elementor_theme_manager->register_all_core_location();
	$elementor_theme_manager->register_location( 'header' );
	$elementor_theme_manager->register_location( 'footer' );
	$elementor_theme_manager->register_location( 'archive' );
}
add_action( 'elementor/theme/register_locations', 'myprefix_register_elementor_locations' );
```

header.blade.php
```
@if ( ! function_exists( 'elementor_theme_do_location' ) || ! elementor_theme_do_location( 'header' ) )
<div class="header">

</div>
@endif
```

footer.blade.php
```
@if ( ! function_exists( 'elementor_theme_do_location' ) || ! elementor_theme_do_location( 'footer' ) )
<div class="footer">

</div>
@endif
```

archive.blade.php
```
@extends('layouts.app')

@section('content')
  @include('partials.page-header')

  @if ( ! function_exists( 'elementor_theme_do_location' ) || ! elementor_theme_do_location( 'archive' ) )

  @if (!have_posts())
    <div class="alert alert-warning">
      {{ __('Sorry, no results were found.', 'sage') }}
    </div>
    {!! get_search_form(false) !!}
  @endif

  @endif

  @while(have_posts()) @php the_post() @endphp
    @include('partials.content-search')
  @endwhile

  {!! get_the_posts_navigation() !!}
@endsection
```

**Comment by gitgudcameron** [comment]
Thanks @MichaelRise. You're a legend. This fixed the issue 👍 I've been trying to solve this for nearly a year now on multiple forums, even had Elementor support login to try to solve it. I assumed $elementor_theme_manager->register_all_core_location() was the same as registering header, footer and archive separately but I was wrong. I appreciate the help!

**Comment by jvdburgh** [comment]
Thx @MichaelRise! 

For future reference, you don't need to add `archive.blade.php`. All you need is this in your `functions.php`:

`  $elementor_theme_manager->register_location('archive');`

`functions.php` snippet
```
/**
 * Register locations for Elementor (https://developers.elementor.com/theme-locations-api/registering-locations/)
 */
function theme_prefix_register_elementor_locations($elementor_theme_manager)
{
    $elementor_theme_manager->register_all_core_location();
    $elementor_theme_manager->register_location('archive');
}
```

So you can actually keep the `register_all_core_location()` in there for the footer, header, 404 and single. Not sure if those are (all) the core locations tho.
